core/src/test/java/org/apache/calcite/tools/FrameworksTest.java

```diff
@@ -292,6 +293,43 @@ private void checkTypeSystem(final int expected, FrameworkConfig config) {
         });
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3228">[CALCITE-3228]
+   * Error while applying rule ProjectScanRule:interpreter</a>
+   *
+   * <p>This bug appears under the following conditions:
+   * 1) have an aggregate with group by and multi aggregate calls.
+   * 2) the aggregate can be removed during optimization.
```

Comment:
The aggregate can't be removed in this test case, right?

Reply:
Why?

Reply:
Because the query evaluates the max/min value in every group, not globally.

Reply:
for the query in my test case which is `select min(id) as mi, max(id) as ma from mytable where id=1 group by id`, there is only on group.

Reply:
You can't remove it if it is empty relation.

Reply:
I believe that `select min(id) as mi, max(id) as ma from mytable where id=1 group by id;`
is equal to
`select id as mi, id as ma from mytable where id=1;` Am i right?

Reply:
The first query returns "NULL, NULL" if there's no tuples with id equals 1, while the 2nd query returns empty result set. They are different.

Reply:
I don't think so. Both the first query and second query return empty result set as following.
```
test=# select * from tmp2;
 id | key
----+-----
  1 |   1
(1 row)

test=# select min(id) from tmp2 where key = 2 group by key;
 min
-----
(0 rows)

test=# select id from tmp2 where key = 2;
 id
----
(0 rows)
```

Reply:
I don't know which version you used, but here is the document -

https://www.postgresql.org/docs/8.2/functions-aggregate.html

"It should be noted that except for count, these functions return a null value when no rows are selected. In particular, sum of no rows returns null, not zero as one might expect. The coalesce function may be used to substitute zero for null when necessary."

Reply:
yes, you are right that result is the same if you add "group by" clause. But anyway the point here is not about aggregation removal. You can even repro this issue with just "select id as id1, id as id2 from mytable where id=1 group by id;".

Reply:
Yes, you can find the same test in `RelBuilderTest.java`.